app/code/Magento/InventorySales/Model/IsProductSalableForRequestedQtyCondition/IsSalableWithReservationsCondition.php

```diff
@@ -84,13 +84,22 @@ public function execute(string $sku, int $stockId, float $requestedQty): Product
             return $this->productSalableResultFactory->create(['errors' => $errors]);
         }
 
+        if(!(bool)$stockItemData['is_salable']){
+            $stockItemData['quantity'] = 0;
```

Comment:
for consistency you need to use constants:
`GetStockItemDataInterface::QUANTITY` for `quantity` and 
`GetStockItemDataInterface::IS_SALABLE` for `is_salable`